test/unit/cmap/connection.test.ts

```diff
@@ -137,6 +157,89 @@ describe('new Connection()', function () {
     });
   });
 
+  describe('#onMessage', function () {
+    context('when the connection is a monitoring connection', function () {
+      let queue: Map<number, OperationDescription>;
+      let driverSocket: FakeSocket;
+      let connection: Connection;
+
+      beforeEach(function () {
+        driverSocket = sinon.spy(new FakeSocket());
+        // @ts-expect-error: driverSocket does not fully satisfy the stream type, but that's okay
+        connection = sinon.spy(new Connection(driverSocket, connectionOptionsDefaults));
+        connection.isMonitoringConnection = true;
+        const queueSymbol = getSymbolFrom(connection, 'queue');
+        queue = connection[queueSymbol];
+      });
+
+      context('when requestId/responseTo do not match', function () {
+        let callbackSpy;
+        const document = { ok: 1 };
+
+        beforeEach(function () {
+          callbackSpy = sinon.spy();
+          // Create the operation description.
+          const operationDescription: OperationDescription = {
+            requestId: 1,
+            cb: callbackSpy
+          };
+
+          // Stick an operation description in the queue.
+          queue.set(1, operationDescription);
+          // Emit a message that won't match the existing operation description.
+          const msg = generateOpMsgBuffer(document);
+          const msgHeader: MessageHeader = {
+            length: msg.readInt32LE(0),
+            requestId: msg.readInt32LE(4),
+            responseTo: msg.readInt32LE(8),
+            opCode: msg.readInt32LE(12)
+          };
+          const msgBody = msg.subarray(16);
+
+          const message = new BinMsg(msg, msgHeader, msgBody);
+          connection.onMessage(message);
+        });
+
+        it('calls the operation description callback with the document', function () {
+          expect(callbackSpy).to.be.calledWith(undefined, document);
```

Comment:
This would fail before the changes as the early would turn would cause the callback never to be called.